frostsnap_core/src/coordinator.rs

```diff
@@ -1422,6 +1524,48 @@ impl FrostCoordinator {
     pub fn cancel_all_keygens(&mut self) {
         self.pending_keygens.clear()
     }
+
+    pub fn knows_about_share(
+        &self,
+        device_id: DeviceId,
+        access_structure_ref: AccessStructureRef,
+        index: PartyIndex,
+    ) -> bool {
+        let already_got_under_key = self
+            .keys
+            .get(&access_structure_ref.key_id)
+            .and_then(|coord_key| {
+                let access_structure_id = access_structure_ref.access_structure_id;
+                Some(
```

Comment:
nit: Don't need `Some` here,
```
      let already_got_under_key = self
          .keys
          .get(&access_structure_ref.key_id)
          .and_then(|coord_key| {
              let access_structure_id = access_structure_ref.access_structure_id;
              coord_key
                  .get_access_structure(access_structure_id)
                  .map(|access_structure| access_structure.contains_device(device_id))
          })
          .unwrap_or(false);
```
but pretty much identical

Reply:
Not sure what the code you're referring to was at the time. Take a look at the latest version.